wasm/store_test.go

```diff
@@ -52,3 +54,45 @@ func TestBuildFunctionInstances_FunctionNames(t *testing.T) {
 	// We expect unknown for any functions missing data in the NameSection
 	require.Equal(t, []string{"unknown", "two", "unknown", "four", "five"}, names)
 }
+
+func TestValidateAddrRange(t *testing.T) {
+	name := "test"
+	s := NewStore(nil)
+	mi := s.getModuleInstance(name)
+
+	m := &Module{
+		MemorySection: []*LimitsType{{Min: 100}},
+	}
+
+	_, err := s.buildMemoryInstances(m, mi)
+	require.NoError(t, err)
+
+	require.True(t, mi.Memory.ValidateAddrRange(uint32(0), uint64(0)))
+	require.True(t, mi.Memory.ValidateAddrRange(uint32(0), uint64(100*PageSize)))
+	require.False(t, mi.Memory.ValidateAddrRange(uint32(0), uint64(100*PageSize+1)))
+	require.False(t, mi.Memory.ValidateAddrRange(uint32(1), uint64(100*PageSize)))
+	require.False(t, mi.Memory.ValidateAddrRange(uint32(100*PageSize), uint64(0)))
+}
+
+func TestPutUint32(t *testing.T) {
+	name := "test"
+	s := NewStore(nil)
+	mi := s.getModuleInstance(name)
```

Comment:
same comment for overkiling

Reply:
Fixed it so that it builds an ad-hoc MemoryInstance from scratch instead of calling buildMemoryInstances.